### PR TITLE
Add update permission on GroupPermission UserGroups

### PR DESCRIPTION
**Pull request by Sheikah45**
Allows updating group permissions tied to https://github.com/FAForever/faf-moderator-client/pull/135

### CODE REVIEW

**Review by Brutus5000 — COMMENTED** [review]
Hmm I would have expected that we need @Update permissions on user_group_assignment and group_permission_assignment and  not on the GroupPermission class  relationship

**Comment by Sheikah45** [comment]
There is no individual classes for the assignments though. They are propagated directly to the userGroup and groupPermission objects as far as I can tell. And since the userGroup is the owning entity any changes there get cascaded to the groupPermission.

Also the assignments would already be covered by the @Update on userGroup. I could be missing something though.

**Comment by Sheikah45** [comment]
It is because of the 
```
  @ManyToMany(cascade = {
    CascadeType.PERSIST,
    CascadeType.MERGE
  })
```
on the permissions field of UserGroup. So changes to userGroup Permissions change groupPermission which then fail if you don't have write permission.

File: src/inttest/java/com/faforever/api/data/GroupPermissionTest.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  @Test
+  public void canReadPermissionsWithScopeAndRole() throws Exception {
+    mockMvc.perform(get("/data/groupPermission")
+      .with(getOAuthTokenWithTestUser(OAuthScope._READ_SENSIBLE_USERDATA, GroupPermission.ROLE_READ_USER_GROUP)))
+      .andExpect(status().isOk())
+      .andExpect(jsonPath("$.data", hasSize(23)));
+  }
+
+  @Test
+  public void cannotCreateUserNoteWithoutScope() throws Exception {
```
Comment (Brutus5000) [inline src/inttest/java/com/faforever/api/data/GroupPermissionTest.java]: UserNote?

File: src/inttest/java/com/faforever/api/data/GroupPermissionTest.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    mockMvc.perform(post("/data/groupPermission")
+      .with(getOAuthTokenWithTestUser(NO_SCOPE, GroupPermission.ROLE_WRITE_USER_GROUP))
+      .header(HttpHeaders.CONTENT_TYPE, JSON_API_MEDIA_TYPE)
+      .content(testPost))
+      .andExpect(status().isForbidden());
+  }
+
+  @Test
+  public void cannotCreateUserNoteWithoutRole() throws Exception {
```
Comment (Brutus5000) [inline src/inttest/java/com/faforever/api/data/GroupPermissionTest.java]: UserNote?

File: src/inttest/java/com/faforever/api/data/GroupPermissionTest.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    mockMvc.perform(post("/data/groupPermission")
+      .with(getOAuthTokenWithTestUser(OAuthScope._ADMINISTRATIVE_ACTION, NO_AUTHORITIES))
+      .header(HttpHeaders.CONTENT_TYPE, JSON_API_MEDIA_TYPE)
+      .content(testPost))
+      .andExpect(status().isForbidden());
+  }
+
+  @Test
+  public void cannotCreateUserNoteWithScopeAndRole() throws Exception {
```
Comment (Brutus5000) [inline src/inttest/java/com/faforever/api/data/GroupPermissionTest.java]: UserNote?

File: src/inttest/resources/sql/prepDefaultData.sql [hunk 4]
```diff
@@ -17,7 +17,8 @@ INSERT INTO user_group (id, technical_name, name_key, parent_group_id, public)
 VALUES (1, 'ADMINISTRATOR', 'administrator', null, true),
        (2, 'MODERATOR', 'moderator', null, true),
        (3, 'faf_server_administrators', 'user_group.faf.server_administrators', null, true),
-       (4, 'faf_moderators_global', 'user_group.faf.moderators.global', null, true);
+       (4, 'faf_moderators_global', 'user_group.faf.moderators.global', null, true),
+       (5, 'faf_illuminati', 'user_group.faf.iiluminati', null, false);
```
Comment (Brutus5000) [inline src/inttest/resources/sql/prepDefaultData.sql]: typo in iiluminati :stuck_out_tongue:

**Review by Brutus5000 — CHANGES_REQUESTED** [review]

File: src/inttest/java/com/faforever/api/data/GroupPermissionTest.java (hunk 1 above)
Comment (Sheikah45) [reply src/inttest/java/com/faforever/api/data/GroupPermissionTest.java]: bad pastaCopy

Comment (Sheikah45) [reply src/inttest/java/com/faforever/api/data/GroupPermissionTest.java]: Fixed

File: src/inttest/java/com/faforever/api/data/GroupPermissionTest.java (hunk 3 above)
Comment (Sheikah45) [reply src/inttest/java/com/faforever/api/data/GroupPermissionTest.java]: Fixed

File: src/inttest/resources/sql/prepDefaultData.sql (hunk 4 above)
Comment (Sheikah45) [reply src/inttest/resources/sql/prepDefaultData.sql]: Fixed

File: src/inttest/java/com/faforever/api/data/GroupPermissionTest.java (hunk 2 above)
Comment (Sheikah45) [reply src/inttest/java/com/faforever/api/data/GroupPermissionTest.java]: Fixed

**Review by Brutus5000 — APPROVED** [review]

**Pull request merged**